work/v1/types.go

```diff
@@ -46,6 +51,56 @@ type ManifestsTemplate struct {
 	Manifests []Manifest `json:"manifests,omitempty"`
 }
 
+type DeleteOption struct {
+	// propagationPolicy can be Orphan or SelectivelyOrphan
+	// SelectivelyOrphan should be rarely used.  It is provided for cases where particular resources is transfering
+	// ownership from one ManifestWork to another or another management unit.
+	// Setting this value will allow a flow like
+	// 1. create manifestwork/2 to manage foo
+	// 2. update manifestwork/1 to selectively orphan foo
+	// 3. remove foo from manifestwork/1 without impacting continuity because manifestwork/2 adopts it.
+	// +optional
+	PropagationPolicy DeletePropagationPolicyType `json:"propagationPolicy,omitempty"`
```

Comment:
There are some discriminated union kubebuilder tags you can use.

Reply:
now that you've switch to CRD v1, you can  use defaulting to set this field to Foreground by default.